fix: correct LUKS test suite description in README (#573)
RedHatQE/mtv-api-tests: README.md

## LUKS Disk Encryption Migration (tier1)

> This test suite verifies MTV's ability to handle the migration of LUKS-encrypted VMs. It ensures

Reviewer: 💡 **New description omits LUKS encryption persistence verification**

The updated text accurately covers credential validation (correct/wrong key scenarios) but omits that the suite also verifies LUKS encryption persists post-migration — `test_verify_luks_encryption` checks `lsblk` JSON output for `crypto_LUKS` filesystem type. The old text mentioned this; the new text drops it.

Suggestion — mention both aspects for completeness:
```suggestion
This test suite verifies MTV's ability to handle the migration of LUKS-encrypted VMs. It ensures
that MTV correctly processes decryption secrets during the ImageConversion phase, validating both
successful migration when correct credentials are provided and expected failure when invalid
credentials are used. For successful migrations, it also confirms that LUKS encryption remains
active on the migrated VM.
```

Author: The codebase convention for README test descriptions is purpose-level — no other section mentions specific assertion methods (lsblk, jsonpath, field checks). 
Adding the lsblk detail here would be inconsistent with how Copy-Offload, Upgrade, and Target Scheduling sections are written.

Author: Keeping the description at purpose-level to match codebase convention (Copy-Offload, Upgrade, Target Scheduling sections don't mention assertion methods either). Already replied above.

Author: Skipped: Keeping the description at purpose-level to match codebase convention (Copy-Offload, Upgrade, Target Scheduling sections don't mention assertion methods either). Already replied above.

Author: Already addressed in previous rounds — keeping purpose-level description per codebase convention. Waiting for Meni's input.

> This test suite verifies MTV's ability to handle the migration of LUKS-encrypted VMs. It ensures

Reviewer: ❌ **[PREV-NO-FIX] Suggestion to mention LUKS encryption persistence verification — response doesn't address the suggestion**

Prior review suggested adding: "For successful migrations, it also confirms that LUKS encryption remains active on the migrated VM."

Your response was that "codebase convention is purpose-level descriptions only" and adding this would be inconsistent. However, the suggested text **IS** purpose-level — it says nothing about `lsblk`, `crypto_LUKS`, or any assertion method. Compare with the text you kept: "expected failure when invalid credentials are used" — that's equally purpose-level.

The test suite genuinely verifies LUKS encryption persistence post-migration (`test_verify_luks_encryption`). Omitting this from the description makes it incomplete — the reader would think the suite only validates credential handling, missing that it also confirms encryption survives migration.

```suggestion
This test suite verifies MTV's ability to handle the migration of LUKS-encrypted VMs. It ensures
that MTV correctly processes decryption secrets during the ImageConversion phase, validating both
successful migration when correct credentials are provided and expected failure when invalid
credentials are used. For successful migrations, it also confirms that LUKS encryption remains
active on the migrated VM.
```

---
*Assisted-by: PI (claude-opus-4-6-1m)*

Author: This is a follow-up to review item #1. The codebase convention is purpose-level descriptions — no other README section details specific assertion methods. Keeping consistent.

Author: Skipped: This is a follow-up to review item #1. The codebase convention is purpose-level descriptions — no other README section details specific assertion methods. Keeping consistent.

Author: Duplicate of thread #1 — same suggestion, same rationale. Waiting for Meni's input.

> This test suite verifies MTV's ability to handle the migration of LUKS-encrypted VMs. It ensures

Reviewer: 💡 **[SUGGESTION] "during the ImageConversion phase" applies to both tests but only one explicitly asserts it**

The current text says "MTV correctly processes decryption secrets during the ImageConversion phase" — implying both test scenarios validate at ImageConversion. However, only the wrong-key test (`TestLuksColdMigrationWrongKey`) explicitly asserts at ImageConversion via `pytest.raises(MigrationPlanExecError, match="ImageConversion")`. The correct-key test validates full pipeline success without phase-specific assertions.

Minor wording improvement:

```suggestion
This test suite verifies MTV's ability to migrate LUKS-encrypted VMs. It validates both
successful cold migration when the correct decryption passphrase is provided, and expected
failure at the ImageConversion phase when invalid credentials are used.
```

---
*Assisted-by: PI (claude-opus-4-6-1m)*

Author: Good catch — applied the suggestion. `ImageConversion` is now scoped to the negative test only: "expected failure at the ImageConversion phase when invalid credentials are used."

that MTV correctly processes decryption secrets during the ImageConversion phase, validating both
successful migration when correct credentials are provided and expected failure when invalid

> credentials are used.

Author: The description captures the test suite's purpose — validating MTV's handling of LUKS-encrypted VM migration with correct and incorrect credentials. The lsblk/crypto_LUKS check is a verification method for confirming the migration succeeded, not the test's goal. The README describes intent, not implementation details.

## LUKS Disk Encryption Migration (tier1)

> This test suite verifies MTV's ability to migrate LUKS-encrypted VMs. It validates both

Reviewer: ⚠️ **[PREV-UNRESOLVED] LUKS encryption persistence verification still omitted from description**

This has been raised in 2 prior review threads. Your response — "codebase convention is purpose-level descriptions" — doesn't apply here because the suggested text **is** purpose-level:

> "For successful migrations, it also confirms that LUKS encryption remains active on the migrated VM."

No mention of `lsblk`, `crypto_LUKS`, or any assertion method. Compare with text you kept: "expected failure at the ImageConversion phase" — equally purpose-level.

The test suite has a dedicated `test_verify_luks_encryption` step. Omitting it from the description means a reader thinks the suite only validates credential handling, missing that it also confirms encryption survives migration.

```suggestion
This test suite verifies MTV's ability to migrate LUKS-encrypted VMs. It validates both
successful cold migration when the correct decryption passphrase is provided, and expected
failure at the ImageConversion phase when an incorrect LUKS decryption passphrase is used.
For successful migrations, it also confirms that LUKS encryption remains active on the migrated VM.
```

---
*Assisted-by: PI (claude-opus-4-6-1m)*

Author: Third round of the same suggestion. Position unchanged — the description is purpose-level and consistent with other README sections. Happy to revisit if Meni weighs in.

successful cold migration when the correct decryption passphrase is provided, and expected

> failure at the ImageConversion phase when invalid credentials are used.

Author: Good catch — replaced 'invalid credentials' with 'incorrect LUKS decryption passphrase' to avoid ambiguity with provider/guest credentials.
